src/Hedgehog/Gen.fs

```diff
@@ -234,42 +256,32 @@ module Gen =
     // Combinators - Conditional
     //
 
-    /// More or less the same logic as suchThatMaybe from QuickCheck, except
-    /// modified to ensure that the shrinks also obey the predicate.
-    let private tryFilterRandom (p : 'a -> bool) (r0 : Random<Tree<'a>>) : Random<Option<Tree<'a>>> =
+    /// Tries to generate a value that satisfies a predicate.
+    let tryFilter (p : 'a -> bool) (g : Gen<'a>) : Gen<'a option> =
         let rec tryN k = function
             | 0 ->
-                Random.constant None
+                constant None
             | n ->
-                let r = Random.resize (2 * k + n) r0
-                Random.bind r <| fun x ->
-                    if p (Tree.outcome x) then
-                        Tree.filter p x |> Some |> Random.constant
+                let g' = resize (2 * k + n) g
+                bind g' <| fun x ->
+                    if p x then
+                        constant (Some x)
```

Comment:
It looks like the actual filtering part was removed. (?)

Reply:
The filter part is still there, it's this part I believe (267-270):

```fsharp
if p x then
    constant (Some x)
else
    tryN (k + 1) (n - 1)
```

Reply:
I agree with @moodmosaic.  Most of the changes in these lines are from renames or from merging modules.  But where did the call to `Tree.filter` go?

If not calling `Tree.filter` is a bug, then we should make a test for it.

Reply:
I agree with @TysonMN, it'd be nice with a test.

Reply:
I see what happened here. There were a few `tryFilter*` functions, `tryFilterRandom` and `tryFilter`. When `Random` was removed, the `tryFilterRandom` function was also removed. The code is close enough that the diff shows them as the same function that changed, when this isn't the case.